Домашнее задание спринт 14

Nataliya2288 opened the pull request.

Nataliya2288 committed:
Домашнее задание спринт 14

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker/AudioPlayerActivity.kt:
```diff
@@ -11,6 +15,31 @@ import java.text.SimpleDateFormat
+
+    private var playerState = STATE_DEFAULT
+
+    private var mediaPlayer = MediaPlayer()
```
```suggestion
    private val mediaPlayer = MediaPlayer()
```
Плеер не пересоздаётся в рамках активити, поэтому можно сделать val

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker/AudioPlayerActivity.kt:
```diff
@@ -11,6 +15,31 @@ import java.text.SimpleDateFormat
+    private val cycleRunnable = object : Runnable {
+        override fun run() {
+            val formattedTime =
+                SimpleDateFormat("mm:ss", Locale.getDefault()).format(mediaPlayer.currentPosition)
```
Сейчас на каждый вызов функции будет создаваться новый объект `SimpleDateFormat("mm:ss", Locale.getDefault())`, что не очень хорошо, этот объект можно создать лениво один раз и закешировать на уровне класса: `private val dateFormat by lazy { SimpleDateFormat("mm:ss", Locale.getDefault()) }`

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker/AudioPlayerActivity.kt:
```diff
@@ -35,38 +68,99 @@ class AudioPlayerActivity : AppCompatActivity() {
+            } else {
+                collectionName.text = "n/a"
             }
+            year.text = track.releaseDate.split("-", limit = 2)[0]
```
Здесь лучше сделать проверку на длину строки, чтобы не выбросился эксепшн

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -27,15 +30,21 @@ import retrofit2.converter.gson.GsonConverterFactory
 
+    private var isClickAllowed = true
+    private val handler = Handler(Looper.getMainLooper())
+    private val searchRunnable = Runnable { search() }
```
Этот раннабл лучше очищать в методе `onDestroy` аналогично тому, как ты очищаешь `cycleRunnable` в `AudioPlayerActivity`

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -212,33 +227,59 @@ class SearchActivity : AppCompatActivity() {
+
+                        when (response.code()) {
+                            200 -> {
+                                if (response.body()?.tracks?.isNotEmpty() == true) {
+                                    tracks.clear()
+                                    tracks.addAll(response.body()?.tracks!!)
```
`response.body()?.tracks` можно вынести в отдельную переменную, чтобы избежать лишних вызовов операторов `?` и `!!`

Nataliya2288 committed:
Корректировка домашнего задания спринт 14
 1.Дочернему элементу ProgressBar добавлне margintop
 2.Отцентрировала при помощи  android:layout_gravity="center_horizontal|top"
 3. ProgressBar находится над всеми виджетами.Каждый виджет настроен на видимость/скрытие с помощью параметра android:visibility. Виджеты с информацией о не найденных данных (not_found_widget), плохом соединении (bad_connection_widget) и истории поиска (history_widget) начально скрыты (android:visibility="gone"), и могут отображаться в соответствии с логикой поиска.

Nataliya2288 committed:
Корректировка домашнего задания спринт 14
Спрятаны : список треков, кнопка, текст "вы искали", картики "заглушки"
